### ISSUE TITLE
Unable to recognize Coefficients

### ISSUE THREAD

**Issue by tushar-mbww** [paul-buerkner/brms#252]
Hi @paul-buerkner ,

This may have been solved earlier , but I wasn't able to find the solution. I'm relatively new to BRMS and was able to run linear multivariate model successfully. 

I'm now trying to run a non-linear model. The non-linear formula is defined b me. Here's part of the code:

list_of_priors <- c(
                    set_prior("normal(0.005,3)",nlpar = "alpha",lb = 0.0001,ub = 5),
                    set_prior("normal(0.005,3)",nlpar = "beta_direct_buy",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_fb",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_gdn",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_lkdin",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_mob",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_bing",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_mob",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_brok",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_google",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_search",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_disp",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_olv ",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "beta_tw",lb = 0.000001,ub = 0.1),
                    set_prior("normal(0.005,3)",nlpar = "power_direct_buy",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_fb",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_gdn",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_lkdin",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_mob",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_bing",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_mob",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_brok",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_google",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_search",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_disp",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_olv ",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "power_tw",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_direct_buy",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_fb",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_gdn",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_lkdin",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_mob",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_bing",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_mob",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_brok",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_google",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_search",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_disp",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_olv ",lb = 0.25,ub = 0.75),
                    set_prior("normal(0.005,3)",nlpar = "ads_tw",lb = 0.25,ub = 0.75)
                  )

fit <- brm(
           cbind(Awareness_LN)  ~ 
             alpha + 
             (exp(beta_direct_buy * (direct_buy/1)^power_direct_buy) - 1) / (1 -ads_direct_buy) + 
             (exp(beta_fb * (fb/1)^power_fb) - 1) / (1 -ads_fb) + 
             (exp(beta_gdn * (gdn/1)^power_gdn) - 1) / (1 -ads_gdn) + 
             (exp(beta_lkdin * (lkdin/1)^power_lkdin) - 1) / (1 -ads_lkdin) + 
             (exp(beta_mob * (mob/1)^power_mob) - 1) / (1 -ads_mob) + 
             (exp(beta_bing * (bing/1)^power_bing) - 1) / (1 -ads_bing) + 
             (exp(beta_mob * (mob/1)^power_mob) - 1) / (1 -ads_mob) + 
             (exp(beta_brok * (brok/1)^power_brok) - 1) / (1 -ads_brok) + 
             (exp(beta_google * (google/1)^power_google) - 1) / (1 -ads_google) + 
             (exp(beta_search * (search/1)^power_search) - 1) / (1 -ads_search) + 
             (exp(beta_disp * (disp/1)^power_disp) - 1) / (1 -ads_disp) + 
             (exp(beta_olv  * (olv /1)^power_olv ) - 1) / (1 -ads_olv ) + 
             (exp(beta_tw * (tw/1)^power_tw) - 1) / (1 -ads_tw)
           , 
           family = "gaussian",
           data = Modelfile_final,
           prior = list_of_priors,
           init_r = 0.0001,
           control = list(adapt_delta = 0.95),
           chains = 5,
           warmup = 1000,
           iter = 3000,
           nl = TRUE
          )

I get the following error:
Error: The following variables are missing in 'data'
![image](https://user-images.githubusercontent.com/31015611/29286171-14a1cf52-80ff-11e7-8295-196717c85b97.png)

Basically, it seems that it's looking at the coefficient and trying to find it as a variable. 
Please note that for every term , I'm trying to estimate 3 separate parameters (Beta , Power & decay)

Not sure if that's what's causing the trouble. Please do have a look when you have a moment.

**Comment by paul-buerkner** [comment]
You did not tell `brm` that these should be non-linear parameters. Please see `?brmsformula` as well as `vignette("brms_multilevel")` and `vignette("brms_nonlinear")` for explanation of the brms non-linear syntax.
